test.js

```diff
@@ -69,10 +73,105 @@ test('it supports multiple arguments', t => {
   source.emit(1, 'qu');
 
   t.deepEqual(
-    sink.getReceivedData(), 
+    sink.getReceivedData(),
     ['foo','bar','baz','qu'],
     'sink gets seed and subsequent data'
   );
 
   t.end();
 });
+
+test('it queues sync completion', t => {
+  const seededSrc = startWith('a', 'b', 'c')(of('d'));
+
+  const sink = makeMockCallbag();
+
+  seededSrc(0, sink);
+
+  t.deepEqual(
+    sink.getMessages().slice(1),
+    [
+      [1, 'a'],
+      [1, 'b'],
+      [1, 'c'],
+      [1, 'd'],
+      [2, undefined],
+    ],
+    'sink gets data in the correct order'
+  );
+
+  t.end();
+});
+
+test('it doesn\'t request data when receiving uknown type', t => {
+  const source = makeMockCallbag(true);
+  const seedWithFoo = startWith('foo');
+  const seededSource = seedWithFoo(source);
+
+  const autoPull = () => {};
+  forEach(autoPull)(seededSource);
```

Comment:
Now that we've brought in `callbag-pipe`, perhaps pipify this too?

As you observed I haven't been using it in other tests, but I've been meaning to. At least when we have more than 3 steps, which feels like a good threshold.